src/index.ts

```diff
@@ -10,7 +10,13 @@ import { SessionTrackingState } from './models/slices/sessiontracking';
 export * from '@yext/answers-core';
 export * from './utils/filter-creators';
 
-type HeadlessConfig = AnswersConfig & {
+/**
+ * The configuration for an AnswersHeadless instance.
+ */
+export type HeadlessConfig = AnswersConfig & {
+  /**
+   * The ID of the AnswersHeadless instance.
+   */
   headlessId?: string
```

Comment:
Maybe we should add a remark that this id cannot be the same as DEFAULT_HEADLESS_ID?

Reply:
Added! I didn't reference DEFAULT_HEADLESS_ID in the remark directly because we aren't planning on exporting that constant.

Reply:
hmm I feel like if we don't reference it, then it's a little vague and user wouldn't really know what can't be use. If we do want to add this remark, I think it would be better to export that variable and mark it with @/internal. @cea2aj what do you think?

Reply:
I agree that it would be good to include the value of the variable which is reserved. We could either mention that the reserved ID is 'main'. There is a risk that we update the ID in the constants file and forget to update it here though. For that reason, I'm in favor of exporting that constant and linking to it. We can't mark it as internal or else API extractors wouldn't include a doc page for it. I think it's reasonable to consider it a public part of the interface since changes to the value affects which headlessIds can be used by consumers of the library

Reply:
ah i didn't realize internal tag prevent api extractor from creating a doc page. I agree, yea let's export the constant and link it in the headless config

Reply:
I'll preface this by saying I don't feel too strongly either way. In my opinion, we don't need to export the `DEFAULT_HEADLESS_ID`. Linking to it feels like overkill a bit. A reader of the documentation needs to know what the reserved value is. By linking to the variable itself, wouldn't we be making them do an extra hop to get to that info? They'd need to click into the link.

Internally, we know that we can't change this identifier already. I'm not worried about someone attempting that.

Reply:
can we change this from a string type to maybe some typescript type that is "any string but our special string"

Reply:
I think it would be good to either make the constant public and link to it in the docs, or explicitly say in the docs what the value of the reserved ID is (although this may be a problem if we update the constant and forget to update the docs as Connor mentioned). Even if we made a type to check that, the user would still need to know which value is reserved by exporting the constant, right?

Reply:
yeah I'm cool with exporting it! I think the excluded string type is actually a bit harder than I thought so we can skip for now
https://stackoverflow.com/questions/51442157/type-for-every-possible-string-value-except

Reply:
Updated to link directly to DEFAULT_HEADLESS_ID, which I will export in the API extractor PR